server/etcdserver/apply.go

```diff
@@ -336,7 +336,7 @@ func (a *applierV3backend) Range(ctx context.Context, txn mvcc.TxnRead, r *pb.Ra
 	resp.Header = &pb.ResponseHeader{}
 
 	if txn == nil {
-		txn = a.s.kv.Read(trace)
+		txn = a.s.kv.Read(true, trace)
```

Comment:
boolean are harming readability a lot. Please make it an enum:

```
type ConcurrentTxMode bool
const (
  // ... expensive at creation, but ...
  CONCURRENT / SHARED_BUFFER = ReadTxMode(true)
  // ... 
  LAZY = ReadTxMode(false)
)
```

Reply:
I am using uint32 here now just in case later we may need more choices.